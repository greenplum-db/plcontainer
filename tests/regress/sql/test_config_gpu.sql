-- start_ignore
\! plcontainer runtime-add -r plc_python3_shared_gpu -i python39.alpine -l python3
\! plcontainer runtime-backup -f test_config_gpu.xml
\! ./test_assign_gpu.py --file ./test_config_gpu.xml -i --runtime plc_python3_shared_gpu --action all
\! plcontainer runtime-restore -f test_config_gpu.xml
\! rm test_config_gpu.xml
-- end_ignore

create function with_conf_gpu() returns int as $$
  # container: plc_python3_shared_gpu
  return 0
$$ LANGUAGE plcontainer;

-- gpu=all means assign all available to this runtime. and not emit error when the GPU is not exist
select * from with_conf_gpu();

drop function with_conf_gpu();

-- start_ignore
\! plcontainer runtime-delete -r plc_python3_shared_gpu
-- end_ignore
