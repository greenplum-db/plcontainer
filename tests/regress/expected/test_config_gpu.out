-- start_ignore
\! plcontainer runtime-add -r plc_python3_shared_gpu -i python39.alpine -l python3
20220628:18:21:44:133476 plcontainer:fluorite:sa-[INFO]:-Distributing file plcontainer_configuration.xml to all locations...
20220628:18:21:44:133476 plcontainer:fluorite:sa-[INFO]:-Configuration has changed. Run "select * from plcontainer_refresh_config" in open sessions. New sessions will get new configuration automatically.
\! plcontainer runtime-backup -f test_config_gpu.xml
\! ./test_assign_gpu.py --file ./test_config_gpu.xml -i --runtime plc_python3_shared_gpu --action all
\! plcontainer runtime-restore -f test_config_gpu.xml
20220628:18:21:45:133734 plcontainer:fluorite:sa-[INFO]:-Distributing file plcontainer_configuration.xml to all locations...
20220628:18:21:45:133734 plcontainer:fluorite:sa-[INFO]:-Configuration has changed. Run "select * from plcontainer_refresh_config" in open sessions. New sessions will get new configuration automatically.
\! rm test_config_gpu.xml
-- end_ignore
create function with_conf_gpu() returns int as $$
  # container: plc_python3_shared_gpu
  return 0
$$ LANGUAGE plcontainer;
-- gpu=all means assign all available to this runtime. and not emit error when the GPU is not exist
select * from with_conf_gpu();
 with_conf_gpu 
---------------
             0
(1 row)

drop function with_conf_gpu();
-- start_ignore
\! plcontainer runtime-delete -r plc_python3_shared_gpu
20220628:18:21:46:134038 plcontainer:fluorite:sa-[INFO]:-Distributing file plcontainer_configuration.xml to all locations...
20220628:18:21:46:134038 plcontainer:fluorite:sa-[INFO]:-Configuration has changed. Run "select * from plcontainer_refresh_config" in open sessions. New sessions will get new configuration automatically.
-- end_ignore
